python/python/lance/torch/data.py

```diff
@@ -234,6 +243,10 @@ def __init__(
         self._to_tensor_fn = to_tensor_fn
         self._hf_converter = None
 
+        self._blob_columns = self._blob_columns()
+        if self._blob_columns:
+            self.with_row_id = True
```

Comment:
Why is this needed?

Reply:
we need row id to call `dataset.take_rows()`